source/fundamentals/crud/read-operations/distinct.txt

```diff
@@ -0,0 +1,117 @@
+========================
+Retrieve Distinct Values
+========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to retrieve distinct values for a
+specified field across a single collection.
+
+Sample Data
+~~~~~~~~~~~
+
+To run the examples in this guide, load the sample data into the
+``ratings`` collection of the ``tea`` database with the following
+snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/distinctValues.go
+   :language: go
+   :dedent:
+   :start-after: begin insert docs
+   :end-before: end insert docs
+
+.. include:: /includes/fundamentals/automatic-db-coll-creation.rst
+
+Each document contains a rating for a type of tea that corresponds to
+the ``type`` and ``rating`` fields.
+
+Distinct
+--------
+
+To retrieve distinct values for a specified field across a single
+collection, pass the following parameters to the ``Distinct()``
+function:
+
+- The field name you want distinct values for 
+- A ``non-nil`` query filter specifying which documents to match
```

Comment:
I'm not sure what you mean by "non-nil" here -- does this just mean that you need to at least pass an empty string, not the `nil` value? If so, would "non-`nil`" be clearer? This might just be my lack of Go experience showing.

Reply:
So in Go, `nil` is actually a value and it represents something (like pointers and types) uninitialized/its zero value. I'm unsure if non-`nil` is more accurate, but I'll jot this note down, follow up with an engineer and make it part of our clean up tasks once all CRUD pages are complete.